src/main/java/com/takipi/oss/storage/resources/fs/multifetcher/MultiFetcher.java

```diff
@@ -16,31 +16,75 @@
 public class MultiFetcher {
     
     private static final Logger logger = LoggerFactory.getLogger(MultiFetcher.class);
+    private static long maxCacheSize = 33554432;
     
     private final TaskExecutor taskExecutor;
+    private final static S3Cache cache;
+    
+    static {
+        if (maxCacheSize > 0) {
+            cache = new S3CacheImpl(maxCacheSize);
+        }
+        else {
+            cache = new S3Cache() {
```

Comment:
Create something like S3Cache.NOP_INSTANCE

Reply:
Why don't you like this implementation?